tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/caches/RecordDomainCache.scala

```diff
@@ -0,0 +1,56 @@
+package at.forsyte.apalache.tla.bmcmt.stratifiedRules.aux.caches
+
+import at.forsyte.apalache.tla.bmcmt.smt.SolverContext
+import at.forsyte.apalache.tla.bmcmt.types.CellT
+import at.forsyte.apalache.tla.bmcmt.{ArenaCell, FixedElemPtr, PureArena}
+import at.forsyte.apalache.tla.lir.{SetT1, StrT1}
+import at.forsyte.apalache.tla.types.tla
+
+import scala.collection.immutable.SortedSet
+
+/**
+ * Since we have to create record domains many times, we cache them here.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+class RecordDomainCache(strValueCache: UninterpretedLiteralCache)
+    extends Cache[PureArena, SortedSet[String], (ArenaCell, Seq[ArenaCell])] {
+
+  /**
+   * Given a set of `keys`, returns a tuple `(rArena, (rCell, rCells))`, where:
+   *   - `rCell` is the cell representing the set `keys`
+   *   - `rCells` is s sequence of cells `c_1, c_2,..., c_|keys|`, representing the set contents (e.g. "a", "b", ...).
+   *   - `rArena` is an extension of `arena`, containing all of the above cells, and a relation
+   *     {{{rCell --(has)--> c_1, c_2,..., c_|keys|}}}
```

Comment:
I would suggest to sync the names here with the code below. WDYT?

```suggestion
   * Given a set of `keys`, returns a tuple `(rArena, (setCell, allCells))`, where:
   *   - `setCell` is the cell representing the set `keys`
   *   - `allCells` is s sequence of cells `c_1, c_2,..., c_|keys|`, representing the set contents (e.g. "a", "b", ...).
   *   - `rArena` is an extension of `arena`, containing all of the above cells, and a relation
   *     {{{setCell --(has)--> c_1, c_2,..., c_|keys|}}}
```